eth/stagedsync/stage_execute.go

```diff
@@ -953,3 +957,19 @@ func PruneExecutionStage(s *PruneState, tx kv.RwTx, cfg ExecuteBlockCfg, ctx con
 	}
 	return nil
 }
+
+func tryUpdateForkVersion(cfg *ExecuteBlockCfg, hermezDb *hermez_db.HermezDb) error {
+	if cfg.chainConfig.Fork7Block != nil && cfg.chainConfig.Fork7Block != big.NewInt(0) {
+		return nil
+	}
+
+	log.Debug("Fork7Block not set, getting from db")
+	// TODO is this the best way to get the fork block? or in memory?
+	blockNum, err := hermezDb.GetForkIdBlock(chain.ZkEVMForkId7)
+	if err != nil {
+		return err
+	}
+	cfg.chainConfig.Fork7Block = big.NewInt(0).SetUint64(blockNum)
```

Comment:
maybe we should do the same for other forkIDs too?

Reply:
Yes, but i think it will be refactor the old code MordorBlock -> Fork7Block

Reply:
Done with:https://github.com/0xPolygonHermez/cdk-erigon/pull/52/files#diff-d3e457f59386d03d6f8da457dd7eea62acb91d0e0d2fc140a66bdafcf716bb9fR30-R37